src/lib.rs

```diff
@@ -459,6 +459,33 @@ impl<T> Grid<T> {
         }
     }
 
+    /// Traverse the grid with row and column indexes.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// use grid::*;
+    /// let grid: Grid<u8> = grid![[1,2][3,4]];
+    /// let mut iter = grid.indexed_iter();
+    /// assert_eq!(iter.next(), Some(((0, 0), &1)));
+    /// ```
+    /// 
+    /// Or simply unpack in a `for`  loop:
+    /// 
+    /// ```
+    /// use grid::*;
+    /// let grid: Grid<u8> = grid![[1,2][3,4]];
+    /// for ((row, col), i) in grid.indexed_iter() {
+    ///     println!("value at row {row} and column {col} is: {i}");
+    /// }
+    /// ```
+    /// 
+    pub fn indexed_iter(&self) -> impl Iterator<Item=((usize, usize), &T)> {
+        self.data.iter().enumerate().map(move |(idx, i)| 
+            ((idx / self.cols, idx % self.cols), i)
```

Comment:
- Would this panic when trying to iterate over an empty grid? Devise by zero. Would be good to have a test for that

Reply:
True. Didn't see that. Thx for adding a test